libs2ecore/src/S2EExecutor.cpp

```diff
@@ -1370,15 +1370,28 @@ Executor::StatePair S2EExecutor::forkAndConcretize(S2EExecutionState *state, kle
 
 S2EExecutor::StatePair S2EExecutor::fork(ExecutionState &current, const klee::ref<Expr> &condition,
                                          bool keepConditionTrueInCurrentState) {
+    return doFork(current, &condition, keepConditionTrueInCurrentState);
+}
+
+S2EExecutor::StatePair S2EExecutor::fork(ExecutionState &current) {
+    return doFork(current, nullptr, false);
+}
+
+S2EExecutor::StatePair S2EExecutor::doFork(ExecutionState &current, const klee::ref<Expr> *condition,
+                                           bool keepConditionTrueInCurrentState) {
```

Comment:
I think the user needs to be able to see the state forking message (written by `S2EExecutor::fork()`).

Since the two overloaded fork() shares a lot of similarity, I refactored them into `S2EExecutor::doFork()`, I hope this is okay.

Reply:
Make doFork private.